### ISSUE TITLE
What do you think about Web Components?

### ISSUE THREAD

**Issue by twixie** [sindresorhus/ama#78]
you're mister node module but what do you think about web components? Will you use them? Why?

**Comment by twixie** [comment]
Additional: how about frameworks like x-tags and polymer?

**Comment by sindresorhus** [comment]
tl;dr I love the concept and really want it to succeed, but it has a long way to go before it's something I would use.

I think it's a great idea in theory. I've always wanted to natively be able to do components (modules) on the front-end. Years ago when I used a lot of jQuery plugins I remember dreading having to manually include the JS, then the CSS file, then having to deal with any conflicts. I always imagined some day having the ability to create self-contained components that you just included with one line without having to be aware of its inner semantics, like having to know that it used this CSS file and this JS file.

Unfortunately though, from my limited experience with Web Components, I'm not yet sold on it. There are a lot of things that are not yet figured out. The packaging story is sub-par. The performance by default is really bad (HTTP 2.0 might improve this), so you end up having to use tools like Vulcanize to inline everything. The spec has changed a lot recently. Everything is pretty much in flux right now. I also found the native API somewhat convoluted, but I guess that's what Polymer and similar libs are for.
